### PR TITLE
CLDR-13641 fix bugs and test based on external data

### PR DESCRIPTION
**Pull request by macchiati**
##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13641
- [ ] Updated PR title and link in previous line to include Issue number

Adding tests against NIST data

### CODE REVIEW

**Comment by macchiati** [comment]
It's blocking on the Jira ticket status; just squashed with the magic button but that didn't help...

**Comment by sffc** [comment]
> It's blocking on the Jira ticket status; just squashed with the magic button but that didn't help...

https://unicode-org.atlassian.net/browse/CLDR-13629 is not accepted yet, and it's marked for 38.

File: common/supplemental/units.xml [hunk 1]
```diff
@@ -188,17 +188,17 @@ For terms of use, see http://www.unicode.org/copyright.html
...
         <convertUnit source='electronvolt' baseUnit='kilogram-square-meter-per-square-second' factor='1.602177E-19'/>
         <convertUnit source='dalton' baseUnit='kilogram-square-meter-per-square-second' factor='1.49241808560E-10'/>
         <convertUnit source='joule' baseUnit='kilogram-square-meter-per-square-second'/>
-        <convertUnit source='calorie' baseUnit='kilogram-square-meter-per-square-second' factor='4.184'/>
-        <convertUnit source='british-thermal-unit' baseUnit='kilogram-square-meter-per-square-second' factor='1055.06' systems="ussystem uksystem"/>
-        <convertUnit source='foodcalorie' baseUnit='kilogram-square-meter-per-square-second' factor='4184' systems="ussystem uksystem"/>
-        <convertUnit source='therm-us' baseUnit='kilogram-square-meter-per-square-second' factor='105506000' systems="ussystem"/>
-        
+        <convertUnit source='calorie' baseUnit='kilogram-square-meter-per-square-second' factor='4.184'/> <!-- th -->
+        <convertUnit source='british-thermal-unit' baseUnit='kilogram-square-meter-per-square-second' factor='4.184*2267.96185/9' systems="ussystem uksystem"/> <!-- th -->
```
Comment (sffc) [inline common/supplemental/units.xml]: What does `<!-- th -->` mean?

**Comment by macchiati** [comment]
That's the nist abbreviation for thermochemical (variant)

**Comment by macchiati** [comment]
Whoops, filed against wrong ticket. Will have to adjust.

**Comment by macchiati** [comment]
Could someone approve this?

**Comment by macchiati** [comment]
Thanks, just changed.

**Comment by sffc** [comment]
And it looks like you already did change the ticket number, so we should be good. :)

**Comment by macchiati** [comment]
Thanks!

**Pull request merged**